### PR TITLE
Move Xcode project

### PR DESCRIPTION
**Pull request by tburgin**
* Created a new Xcode project at the at root and removed the x dir.
  * This will make integration of the current code for each component easier.
* Created the next gen `com.google.santa` and `com.google.santa.daemon` targets.
  * Current code is compiled into the respective target.
  * Added basic system extension loading to com.google.santa.
  * `com.google.santa.daemon` actually has 2 targets. One for the raw `santad` binary and another for the `sysx` that embeds `santad`. This enables faster debugging by running `santad` directly in Xcode.
* The rest of the components still need to be added.

### CODE REVIEW

File: Source/Santa/SNTAppDelegate.m [hunk 1]
```diff
@@ -161,4 +175,26 @@ - (void)setupMenu {
   [NSApp setMainMenu:mainMenu];
 }
 
+#pragma mark OSSystemExtensionRequestDelegate
+
+- (OSSystemExtensionReplacementAction)request:(OSSystemExtensionRequest *)request
+                  actionForReplacingExtension:(OSSystemExtensionProperties *)existing
+                                withExtension:(OSSystemExtensionProperties *)ext {
+  LOGD(@"Received request to replace system exension");
```
Comment (russellhancox) [inline Source/Santa/SNTAppDelegate.m]: spelling: extension

File: Source/Santa/SNTAppDelegate.m [hunk 2]
```diff
@@ -161,4 +175,26 @@ - (void)setupMenu {
...
+  return OSSystemExtensionReplacementActionReplace;
+}
+
+- (void)requestNeedsUserApproval:(OSSystemExtensionRequest *)request {
+  LOGD(@"System exension request needs user approval");
```
Comment (russellhancox) [inline Source/Santa/SNTAppDelegate.m]: sp

File: Source/Santa/SNTAppDelegate.m [hunk 3]
```diff
@@ -161,4 +175,26 @@ - (void)setupMenu {
...
+}
+
+- (void)request:(OSSystemExtensionRequest *)request didFailWithError:(NSError *)error {
+  LOGD(@"System exension request did fail: %@", error);
```
Comment (russellhancox) [inline Source/Santa/SNTAppDelegate.m]: sp

File: Source/Santa/SNTAppDelegate.m [hunk 4]
```diff
@@ -161,4 +175,26 @@ - (void)setupMenu {
...
+}
+
+- (void)request:(OSSystemExtensionRequest *)request
+    didFinishWithResult:(OSSystemExtensionRequestResult)result {
+  LOGD(@"System exension request did finish: %ld", (long)result);
```
Comment (russellhancox) [inline Source/Santa/SNTAppDelegate.m]: sp

File: Source/Santa/SNTAppDelegate.m (hunk 1 above)
Comment (tburgin) [reply Source/Santa/SNTAppDelegate.m]: Ooops

**Pull request merged**